src/MetricConversionServlet.java

```diff
@@ -0,0 +1,64 @@
+import java.io.IOException;
+import java.io.PrintWriter;
+
+@javax.servlet.annotation.WebServlet("/calculateMeters")
+public class MetricConversionServlet extends javax.servlet.http.HttpServlet {
+    protected void doPost(javax.servlet.http.HttpServletRequest request, javax.servlet.http.HttpServletResponse response) throws javax.servlet.ServletException, IOException {
+        //Params
+        request.setCharacterEncoding("UTF-8");
+        String meters =         request.getParameter("meters");
+        String centimeters =    request.getParameter("centimeters");
+        String millimeters =    request.getParameter("milimeters");
+
+        //Log to console inputParams
+        System.out.println("Web service request: "  + meters + ";"
+                                                    + centimeters + ";"
+                                                    + millimeters);
+        //Check invalid input
+        int incorrectInputFlag      = 0;
+        if(!meters.isEmpty())       incorrectInputFlag+=1;
+        if(!centimeters.isEmpty())  incorrectInputFlag+=2;
+        if(!millimeters.isEmpty())  incorrectInputFlag+=4;
+
+        //Set result encoding
+        response.setCharacterEncoding("UTF-8");
+        response.setContentType("text/html");
+        PrintWriter writer = response.getWriter();
+
+        /*Error codes
```

Comment:
Widzę że chciałeś zwrócić za każdym razem inną (bardzo szczegółową) informację na temat wprowadzonych danych. Fajnie że to zrobiłeś, ale mogłeś się ograniczyć tylko do poszczególnych pól :) Tak czy siak napracowałeś się.